packages/databyss-ui/editor/state/page/reducer.js

```diff
@@ -128,8 +129,9 @@ const setBlockType = (state, type, _id) => {
 
   switch (type) {
     case 'SOURCE':
-      nextState.sources[nextRefId] = { _id: nextRefId, textValue, ranges }
-
+      const _source = { _id: nextRefId, textValue, ranges }
+      nextState.sources[nextRefId] = _source
+      nextState.newSources = [_source]
```

Comment:
If there's still a source in the q, this will clobber it. I think we want `nextState.newSources.push(_source)`